File: maths/cycle1/fiches-methodes3e/I/Ruby/m0101.rb
#!/usr/bin/env ruby

require_relative 'method01'
require_relative 'corrective'

class Solution < Method01
  include Corrective

  def answering
    puts "Existe-t-il un entier q tel que 180 = 15q ?"
    integer = 180
    divisor = 15
    q = integer / divisor
    if integer % divisor == 0 then
      puts "#{integer} est divisible par #{divisor} :"
      puts "il existe un entier q tel que #{divisor} x q = #{integer} : #{q}"
    else
      puts "#{integer} n'est pas le produit de #{divisor} par un entier q"
    end

    puts ""
    puts "Existe-t-il un entier q tel que 260 = 15q ?"
    integer = 260
    q = integer / divisor
    m = divisor * q
    n = divisor * (q + 1)
    if integer % divisor == 0 then
      puts "#{integer} est divisible par #{divisor} :"
      puts "il existe un entier q tel que #{divisor} x q = #{integer} : #{q}"
    else
      puts "#{integer} n'est pas le produit de #{divisor} par un entier q"
      puts "#{integer} / #{divisor} = #{q}"
      puts "#{divisor} x #{q} = #{m}"
      puts "#{divisor} x #{q+1} = #{n}"
      puts "or #{m} < #{integer} < #{n}"
      puts "et comme #{q} et #{q+1} sont deux entiers consécutifs,"
      puts "alors il n'existe pas d'entier q tel que #{integer} = #{divisor} x q"
    end
  end
end

s = Solution.new
s.exercise_01
bool = true
bool = s.put_up_method?
if bool then
  s.put_up_method
end
s.put_up_correct_version?
s.answering

File: maths/cycle1/fiches-methodes3e/I/Pascal/m0101.pas
program m0101;

{$mode objfpc}{$H+}

uses
    method01, display;

var
    num : Integer = 180;
    divisor : Integer = 15;
    q, m, n : Integer;
    bool : Boolean;

begin
    WriteLn('# Exercice 1 :');
    WriteLn('# -----------');
    WriteLn('# 180 et 260 sont-ils des multiples de 15 ?');
    WriteLn('-----------------------------------------------------------------------------------');
    WriteLn();

    bool := DisplayMethodWording;
    if bool then PutAMethod();

    bool := PutUpCorrectVersion;
    if not bool then exit;

    q := num div divisor; 
    WriteLn('Existe-t-il un entier q tel que 180 = 15q ?');

    if (num mod divisor = 0) then
        begin
            WriteLn(num, ' est divisible par ', divisor, ' :');
            WriteLn('il existe un entier q tel que ', divisor, ' x q = ', num, ' : ', q);
        end
    else
        WriteLn(num, ' n''est pas le produit de ', divisor, ' par un entier q');

    WriteLn;
    WriteLn('Existe-t-il un entier q tel que 260 = 15q ?');
    num := 260;
    q := num div divisor;
    m := divisor * q;
    n := divisor * (q + 1);
    if (num mod divisor = 0) then
        begin
            WriteLn(num, ' est divisible par ', divisor, ' :');
            WriteLn('il existe un entier q tel que ', divisor, ' x q = ', num, ' : ', q);
        end
    else
        begin
            WriteLn(num, ' n''est pas le produit de ', divisor, ' par un entier q');
            WriteLn(num, ' ÷ ', divisor, ' = ', q);
            WriteLn(divisor, ' x ', q, ' = ', m);
            WriteLn(divisor, ' x ', q+1, ' = ', n);
            WriteLn('or ', m, ' < ', num, ' < ', n);
            WriteLn('et comme ', q, ' et ', q+1, ' sont deux entiers consécutifs,');
            WriteLn('alors il n''existe pas d''entier q tel que ', num, ' = ', divisor, ' x q');
        end;

    {$IFDEF WINDOWS}
    ReadLn;
    {$ENDIF}
end.

